lib/floe/workflow.rb

```diff
@@ -119,8 +119,11 @@ def run_nonblock
     def step_nonblock
       return Errno::EPERM if end?
 
+      result = current_state.run_nonblock!
       step_next
-      current_state.run_nonblock!.tap { end_workflow }
+      end_workflow
```

Comment:
think we only want to `step_next ; end_workflow` if the state is successful (return 0)

Reply:
added a diff below showing the issue